src/v/kafka/requests/fetch_request.cc

```diff
@@ -523,37 +530,60 @@ static ss::future<> fetch_topic_partition(
  * response must be reassembled such that the responses appear in these
  * order as the partitions in the request.
  */
-static ss::future<> fetch_topic_partitions(op_context& octx) {
-    return ss::do_with(
-      octx.response_begin(), [&octx](op_context::response_iterator& resp_it) {
-          return ss::do_for_each(
-                   octx.request.cbegin(),
-                   octx.request.cend(),
-                   [&octx, &resp_it](
-                     const fetch_request::const_iterator::value_type& p) {
-                       return fetch_topic_partition(octx, p, resp_it)
-                         .then([&resp_it] { resp_it++; });
-                   })
-            .then([&octx] { octx.reset_context(); });
+static ss::future<> fetch_topic_partitions(ss::lw_shared_ptr<op_context> octx) {
+    auto resp_it = octx->response_begin();
+    std::vector<ss::future<>> fetches;
+    std::transform(
```

Comment:
is this a chance for denial of service requests: requests that we blinding dispatch in parallel to all cores? should we broadcast to all cores to cancel on-going reads when we satisfy the request?

Reply:
I think we should implement throttling, there is no difference between dispatching in parallel to all the cores and doing it sequentially when it comes to denial of service. We can instruct the client to not wait for the response, we will end up having requests dispatched to all the cores.

Reply:
we should use a `ss::semaphore(10)` or some other semaphore. it is different in regards to memory allocation.

Reply:
or rather `ss::semaphore(smp::count)`

Reply:
let's chat about this, I understand that we should keep a semaphore per connection ?

Reply:
do you think we can add it as a separate patch ?

Reply:
we chatted about this; resolving. the idea is to use `sleep_abortable` for a small debounce